core/math/quaternion.cpp

```diff
@@ -313,6 +313,42 @@ Quaternion::Quaternion(const Vector3 &p_axis, real_t p_angle) {
 	}
 }
 
+static bool _try_set_perpendicular_to(real_t p_in_x, real_t p_in_y, real_t p_in_z, real_t &r_out_x, real_t &r_out_y, real_t &r_out_z) {
+	real_t threshold = 1.0f / Math::sqrt(3.0f);
+	real_t abs_x = Math::abs(p_in_x);
+	if (abs_x > threshold + (real_t)CMP_EPSILON) {
+		return false;
+	}
+
+	real_t length = Math::sqrt(1.0f - abs_x * abs_x);
+	r_out_x = 0;
+	r_out_y = p_in_z / length;
+	r_out_z = -p_in_y / length;
+	return true;
+}
+
+Quaternion::Quaternion(const Vector3 &p_v0, const Vector3 &p_v1) {
+	Vector3 c = p_v0.cross(p_v1);
+	real_t d = p_v0.dot(p_v1);
+
+	if (d < -1.0f + (real_t)CMP_EPSILON) {
```

Comment:
I believe the following tells the compiler that one of the two branches is more likely, which can make a difference in performance.
```suggestion
	if (likely(d < -1.0f + (real_t)CMP_EPSILON)) {
```
or
```suggestion
	if (unlikely(d < -1.0f + (real_t)CMP_EPSILON)) {
```